activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -533,8 +533,43 @@ def insert_fixtures(*)
         without_sql_mode("NO_AUTO_VALUE_ON_ZERO") { super }
       end
 
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        iterate_over_results = -> { while raw_connection.next_result; end; }
+
+        with_multi_statements do
+          without_sql_mode("NO_AUTO_VALUE_ON_ZERO") do
+            super(fixture_set, tables_to_delete, &iterate_over_results)
+          end
+        end
+      end
+
       private
 
+        def max_allowed_packet_reached?(current_packet, previous_packet)
+          if current_packet.bytesize > max_allowed_packet
+            raise ActiveRecordError, "Fixtures set is too large #{current_packet.bytesize}. Consider increasing the max_allowed_packet variable."
+          elsif previous_packet.nil?
+            false
+          else
+            (current_packet.bytesize + previous_packet.bytesize) > max_allowed_packet
+          end
+        end
+
+        def max_allowed_packet
+          @max_allowed_packet ||= execute("SELECT @@SESSION.max_allowed_packet").first[0].to_i
+        end
+
+        def with_multi_statements
+          previous_flags = @config[:flags]
+          @config[:flags] = Mysql2::Client::MULTI_STATEMENTS
+          reconnect!
```

Comment:
Why does this require reconnecting? This should just require a call to `mysql_set_server_option` on the C side

Reply:
Nice, wasn't aware of `mysql_set_server_option`. I can't find any reference of it in mysql2 though :(

Reply:
Yeah, looks like we might have to go with this for now, and then see about getting that method added to mysql2 in future.

(Maybe you can get it sneaked in to the [upcoming release](https://github.com/rails/rails/issues/31636).)

Reply:
How about adding `MULTI_STATEMENTS` by default?
We already have `FOUND_ROWS` to make it the same as PostgreSQL's behavior. Related https://github.com/rails/rails/pull/25370#issuecomment-230157091.

https://github.com/rails/rails/blob/fd04c8cbb6627ef65ea728b5026b21bd24471c18/activerecord/lib/active_record/connection_adapters/mysql2_adapter.rb#L12-L20

Reply:
I'd prefer not to have `MULTI_STATEMENTS` by default. It has a lot of gotchas on MySQL which can cause really subtle bugs